lib/rules/newline-per-chained-call.js

```diff
@@ -70,13 +82,16 @@ module.exports = {
                     parent = parent.callee.object;
                 }
 
-                if (depth > ignoreChainWithDepth && callee.property.loc.start.line === callee.object.loc.end.line) {
+                if (depth > ignoreChainWithDepth && astUtils.isTokenOnSameLine(callee.object, callee.property)) {
                     context.report({
                         node: callee.property,
                         loc: callee.property.loc.start,
                         message: "Expected line break before `{{callee}}`.",
                         data: {
                             callee: getPropertyText(callee)
+                        },
+                        fix(fixer) {
+                            return fixer.replaceTextRange([callee.object.range[1], callee.property.range[0]], `\n${getPrefix(callee)}`);
```

Comment:
I think this will work incorrectly if the object or the property is parenthesized. `callee.object.range` doesn't include any surrounding parentheses, so for text like `(foo).bar()`, `callee.object.range` will be just the range of the `foo` token. Similarly, for text like `foo[(bar)]()`, I think the current implementation will produce `foo\n[bar)]`, which is invalid syntax.

You could avoid parentheses by doing something like this:

```js
fix(fixer) {
    const firstTokenAfterObject = sourceCode.getTokenAfter(callee.object, astUtils.isNotClosingParenToken);
    return fixer.insertTextBefore(firstTokenAfterObject, "\n");
}
```